src/v/cluster/types.h

```diff
@@ -372,29 +383,46 @@ struct topic_properties {
     storage::ntp_config::default_overrides get_ntp_cfg_overrides() const;
 
     friend std::ostream& operator<<(std::ostream&, const topic_properties&);
+    friend bool operator==(const topic_properties&, const topic_properties&)
+      = default;
 };
 
 enum incremental_update_operation : int8_t { none, set, remove };
 template<typename T>
 struct property_update {
+    using value_t = property_update<T>;
+    static constexpr serde::version_t redpanda_serde_version = 0;
+    static constexpr serde::version_t redpanda_serde_compat_version = 0;
     T value;
     incremental_update_operation op = incremental_update_operation::none;
 
+    auto serde_fields() { return std::tie(value, op); }
+
     friend bool operator==(const property_update<T>&, const property_update<T>&)
       = default;
 };
 
 template<typename T>
 struct property_update<tristate<T>> {
+    using value_t = property_update<tristate<T>>;
+    static constexpr serde::version_t redpanda_serde_version = 0;
+    static constexpr serde::version_t redpanda_serde_compat_version = 0;
+
     tristate<T> value = tristate<T>(std::nullopt);
     incremental_update_operation op = incremental_update_operation::none;
 
+    auto serde_fields() { return std::tie(value, op); }
+
     friend bool operator==(
       const property_update<tristate<T>>&, const property_update<tristate<T>>&)
       = default;
 };
 
-struct incremental_topic_updates {
+struct incremental_topic_updates
```

Comment:
this struct uses a negative length trick to encode version. I though that we can eliminate this hack when we move to `serde`. What's the plan for this structure? To keep already existing `version_with_data_policy` and `version_with_shadow_indexing`? And moving forward to increment serde version, right?